zipa/resource.py

```diff
@@ -74,29 +75,36 @@ def _prepare_data(self, **kwargs):
         else:
             return kwargs
 
-    def _prepare_entity(self, response):
+    @staticmethod
+    def default_response_handler(response):
         try:
-            json = response.json()
-        except:
-            json = {}
-
-        http_error_msg = ''
+            parsed_response = response.json()
+        except ValueError:
+            parsed_response = {}
 
         if 400 <= response.status_code < 500:
             http_error_msg = '%s Client Error: %s' % (response.status_code,
                                                       response.reason)
-
         elif 500 <= response.status_code < 600:
             http_error_msg = '%s Server Error: %s' % (response.status_code,
                                                       response.reason)
+        else:
+            http_error_msg = None
 
         if http_error_msg:
-            raise HTTPError(http_error_msg, json, response=response)
+            raise HTTPError(http_error_msg, parsed_response, response=response)
+
+        return parsed_response
+
+    def _prepare_entity(self, response):
+        response_handler = self.config.response_handler or self.default_response_handler
+
+        parsed_response = response_handler(response) or {}
```

Comment:
Is a little bit confusing since a response handler implies that everything related to the response is parsed by it. (entity parsing is included.)